datafusion/proto/src/physical_plan/mod.rs

```diff
@@ -1911,6 +1912,30 @@ pub trait PhysicalExtensionCodec: Debug + Send + Sync {
     ) -> Result<Arc<dyn ExecutionPlan>>;
 
     fn try_encode(&self, node: Arc<dyn ExecutionPlan>, buf: &mut Vec<u8>) -> Result<()>;
+
+    // fn try_decode_udf(
+    //     &self,
+    //     name: &str,
+    //     buf: &[u8],
+    // ) -> Result<Arc<ScalarUDF>>;
+
+    // fn try_encode_udf(&self, node: &ScalarUDF, buf: &mut Vec<u8>) -> Result<()>;
+
+    // fn try_decode_udaf(
+    //     &self,
+    //     name: &str,
+    //     buf: &[u8],
+    // ) -> Result<Arc<AggregateUDF>>;
+
+    // fn try_encode_udaf(&self, node: &AggregateUDF, buf: &mut Vec<u8>) -> Result<()>;
+
+    // fn try_decode_udwf(
+    //     &self,
+    //     name: &str,
+    //     buf: &[u8],
+    // ) -> Result<Arc<WindowUDF>>;
+
+    // fn try_encode_udwf(&self, node: &WindowUDF, buf: &mut Vec<u8>) -> Result<()>;
```

Comment:
Plan to finish parts related to `PhysicalExtensionCodec` after `LogicalExtensionCodec`